### PR TITLE
Purge `Registry` struct

### PR DESCRIPTION
**Pull request by sokorototo**
As you noticed the `Registry` struct was completely unnecessary. It was just a thin wrapper around logic that should have been implemented in the `Archive`. Therefore, I purge it and scribe it to the depths of hell and unnecessary complexity. I hope you find my decision beneficial.

### CODE REVIEW

File: vach/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "vach"
-version = "0.1.3"
+version = "0.1.5"
```
Comment (zeskeertwee) [inline vach/Cargo.toml:3]: Shouldn't we only update the version when we release a new version on crates.io?

**Review by zeskeertwee — CHANGES_REQUESTED** [review]

File: vach/Cargo.toml (hunk 1 above)
Comment (sokorototo) [reply vach/Cargo.toml:3]: I plan on updating the spec, this is not supposed to be here

**Comment by zeskeertwee** [comment]
Seems good to me!

**Review by zeskeertwee — APPROVED** [review]

**Comment by sokorototo** [comment]
Just some conflict resolution, opened a different PR

**Pull request closed without merging**